src/monster.cpp

```diff
@@ -1532,55 +1532,39 @@ void monster::explode()
     }
     // Send body parts and blood all over!
     const itype_id meat = type->get_meat_itype();
+    if( meat == "null" ) {
+        return; // Only create chunks if we know what kind to make.
+    }
+    const int num_chunks = type->get_meat_chunks_count();
     const field_id type_blood = bloodType();
     const field_id type_gib = gibType();
-    if( meat != "null" || type_blood != fd_null || type_gib != fd_null ) {
-        // Only create chunks if we know what kind to make.
-        int num_chunks = 0;
-        switch( type->size ) {
-            case MS_TINY:
-                num_chunks = 1;
-                break;
-            case MS_SMALL:
-                num_chunks = 2;
-                break;
-            case MS_MEDIUM:
-                num_chunks = 4;
-                break;
-            case MS_LARGE:
-                num_chunks = 8;
-                break;
-            case MS_HUGE:
-                num_chunks = 16;
-                break;
-        }
 
-        for( int i = 0; i < num_chunks; i++ ) {
-            tripoint tarp( posx() + rng( -3, 3 ), posy() + rng( -3, 3 ), posz() );
-            std::vector<tripoint> traj = line_to( pos(), tarp, 0, 0 );
+    for( int i = 0; i < num_chunks; i++ ) {
+        tripoint tarp( pos() + tripoint( rng( -3, 3 ), rng( -3, 3 ), 0 ) );
```

Comment:
You can add points to tripoints, no need to explicitly specify that 0 z component.